File: src/stores/me.ts
import { defineStore } from "pinia";
import { ref } from "vue";
import { invoke } from "@tauri-apps/api/core";

export interface Insight {
  id: string;
  content: string;
  source: "dialogue" | "manual";
  dialogueId?: string;
  createdAt: string;
}

export interface MyPhilosophy {
  name: string;
  manifesto: string;
  insights: Insight[];
  context: string;
  updatedAt: string;
}

export const useMeStore = defineStore("me", () => {
  const myPhilosophy = ref<MyPhilosophy>({
    name: "我",
    manifesto: "",
    insights: [],
    context: "",
    updatedAt: new Date().toISOString(),
  });
  const loading = ref(false);

  async function loadMyPhilosophy() {
    loading.value = true;
    try {
      myPhilosophy.value = await invoke<MyPhilosophy>("get_my_philosophy");
    } catch (e) {
      console.error("Failed to load my philosophy:", e);
    } finally {
      loading.value = false;
    }
  }

  async function saveMyPhilosophy() {
    myPhilosophy.value.updatedAt = new Date().toISOString();
    await invoke("save_my_philosophy", { data: myPhilosophy.value });
  }

  async function addInsight(content: string, source: "dialogue" | "manual", dialogueId?: string) {
    const insight: Insight = {
      id: crypto.randomUUID(),
      content,
      source,
      dialogueId,
      createdAt: new Date().toISOString(),
    };
    await invoke("add_insight", { insight });
    myPhilosophy.value.insights.push(insight);
  }

  return {
    myPhilosophy,
    loading,
    loadMyPhilosophy,
    saveMyPhilosophy,
    addInsight,
  };
});

File: src-tauri/src/commands/me.rs
use crate::error::AppError;
use crate::storage;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Insight {
    pub id: String,
    pub content: String,
    pub source: String,
    pub dialogue_id: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MyPhilosophy {
    pub name: String,
    pub manifesto: String,
    pub insights: Vec<Insight>,
    pub context: String,
    pub updated_at: String,
}

impl Default for MyPhilosophy {
    fn default() -> Self {
        Self {
            name: "我".into(),
            manifesto: String::new(),
            insights: Vec::new(),
            context: String::new(),
            updated_at: chrono::Utc::now().to_rfc3339(),
        }
    }
}

fn me_path() -> Result<std::path::PathBuf, AppError> {
    let dir = storage::data_dir()?;
    storage::ensure_dir(&dir)?;
    Ok(dir.join("me.json"))
}

#[tauri::command]
pub fn get_my_philosophy() -> Result<MyPhilosophy, AppError> {
    let path = me_path()?;
    Ok(storage::read_json(&path)?.unwrap_or_default())
}

#[tauri::command]
pub fn save_my_philosophy(data: MyPhilosophy) -> Result<(), AppError> {
    let path = me_path()?;
    storage::write_json(&path, &data)
}

#[tauri::command]
pub fn add_insight(insight: Insight) -> Result<(), AppError> {
    let path = me_path()?;
    let mut my_philosophy: MyPhilosophy = storage::read_json(&path)?.unwrap_or_default();
    my_philosophy.insights.push(insight);
    my_philosophy.updated_at = chrono::Utc::now().to_rfc3339();
    storage::write_json(&path, &my_philosophy)
}
